src/trackForMutations.js

```diff
@@ -1,26 +1,35 @@
-export default function trackForMutations(isImmutable, obj) {
-  const trackedProperties = trackProperties(isImmutable, obj);
+export default function trackForMutations(isImmutable, ignore, obj) {
+  const trackedProperties = trackProperties(isImmutable, ignore, obj);
   return {
     detectMutations() {
-      return detectMutations(isImmutable, trackedProperties, obj);
+      return detectMutations(isImmutable, ignore, trackedProperties, obj);
     }
   };
 }
 
-function trackProperties(isImmutable, obj) {
+function trackProperties(isImmutable, ignore = [], obj, path = []) {
   const tracked = { value: obj };
 
   if (!isImmutable(obj)) {
     tracked.children = {};
 
     for (const key in obj) {
-      tracked.children[key] = trackProperties(isImmutable, obj[key]);
+      if (ignore.length && ignore.includes([ ...path, key].join('.'))) {
```

Comment:
You can move the `path.concat(key)` from below to a variable and use it here as well. Also, let's use `indexOf` since `includes` is not yet supported in all browsers and requires a polyfill.